### PR TITLE
Manifest and resources

### LINKED ISSUE #105: Add docs for dockerfile.v0 resource type.
Add documentation for the new dockerfile.v0 resource type. This probably needs a few elements:

1. A change to the manifest documentation so that it is listed there.
2. A tutorial on how to launch JavaScript apps might make reference to it as a mechanism to use to get those JS apps to production via AZD in a container.

---
[Associated WorkItem - 192276](https://dev.azure.com/msft-skilling/Content/_workitems/edit/192276)

### LINKED ISSUE #211: List of supported AZD resources vs manifest output
See: 

https://github.com/Azure/azure-dev/blob/cb86a7d6b0a4956315f876bac32bc82c2f14ad08/cli/azd/pkg/apphost/generate.go#L240-L285

---
[Associated WorkItem - 192275](https://dev.azure.com/msft-skilling/Content/_workitems/edit/192275)

### LINKED ISSUE #216: Document `connectionString` and `inputs` fields on `container.v0` resources in manifest.
Manifest docs need to cover the new `connectionString` field that can be present on `container.v0` resource types and the way that the `inputs` field works in conjunction with that.

---
[Associated WorkItem - 192277](https://dev.azure.com/msft-skilling/Content/_workitems/edit/192277)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Add `dockerfile.v0` example bits to manifest doc.

Issues addressed in this PR:

- Fixes #105
- Fixes #211
- Fixed #216

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/deployment/manifest-format.md](https://github.com/dotnet/docs-aspire/blob/fea0d780b173d20439e2eda8aa7f4e1bcf2c6f9e/docs/deployment/manifest-format.md) | [.NET Aspire manifest format for deployment tool builders](https://review.learn.microsoft.com/en-us/dotnet/aspire/deployment/manifest-format?branch=pr-en-us-221) |

### CODE REVIEW

**Comment by IEvangelist** [comment]
Pulls in bits from #222

File: docs/deployment/manifest-format.md [hunk 1]
```diff
@@ -358,7 +457,17 @@ Example manifest:
...
+      "image": "redis:latest",
+      "bindings": {
+        "tcp": {
+          "scheme": "tcp",
+          "protocol": "tcp",
+          "transport": "tcp",
+          "containerPort": 6379
+        }
+      },
+      "connectionString": "{rediscontainer.bindings.tcp.host}:{rediscontainer.bindings.tcp.port}"
```
Comment (mitchdenny) [inline docs/deployment/manifest-format.md:470]: Should be: `{redis2.bindings.tcp.host}:{redis2.bindings.tcp.port}`

File: docs/deployment/manifest-format.md [hunk 2]
```diff
@@ -384,11 +492,35 @@ Example manifest:
...
+      },
+      "bindings": {
+        "tcp": {
+          "scheme": "tcp",
+          "protocol": "tcp",
+          "transport": "tcp",
+          "containerPort": 1433
+        }
+      },
+      "connectionString": "Server={sqlservercontainer.bindings.tcp.host},{sqlservercontainer.bindings.tcp.port};User ID=sa;Password={sqlservercontainer.inputs.password};TrustServerCertificate=true;",
```
Comment (mitchdenny) [inline docs/deployment/manifest-format.md:512]: Should be `Server={sql2.bindings.tcp.host},{sql2.bindings.tcp.port};User ID=sa;Password={sql2.inputs.password};TrustServerCertificate=true;`

**Comment by mitchdenny** [comment]
The other thing that we need to slot in here somewhere is how the `inputs` field works.

File: docs/deployment/manifest-format.md (hunk 2 above)
Comment (IEvangelist) [reply docs/deployment/manifest-format.md]: Done

File: docs/deployment/manifest-format.md (hunk 1 above)
Comment (IEvangelist) [reply docs/deployment/manifest-format.md]: Done

**Comment by IEvangelist** [comment]
> The other thing that we need to slot in here somewhere is how the `inputs` field works.

Hi @mitchdenny - see 29248b0 for the addition of the `inputs` field.

File: docs/deployment/manifest-format.md [hunk 3]
```diff
@@ -398,6 +556,104 @@ Example manifest:
...
+
+Example code:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddMySql("mysql1");
+
+builder.AddMySql("mysql2")
+       .AddDatabase("shipping");
```
Comment (adegeo) [inline docs/deployment/manifest-format.md:611]: Couldn't you chain this?

```suggestion
builder.AddMySql("mysql1")
       .AddMySql("mysql2")
       .AddDatabase("shipping");
```

**Review by adegeo — APPROVED** [review]

File: docs/deployment/manifest-format.md (hunk 3 above)
Comment (IEvangelist) [reply docs/deployment/manifest-format.md:611]: No, since the `AddMySql` cannot be chained on the return type from itself.

**Comment by mitchdenny** [comment]
I think this covers it for now. Thanks.

**Pull request merged**